main.js

```diff
@@ -176,6 +183,32 @@ if (buildStatic) {
     return;
 }
 
+function getFileHeaders(headerPath) {
+    if (headerPath === 'NO_HEADERS_FILE') {
+        return {};
+    }
+
+    try {
+        return JSON.parse(fs.readFileSync(`${headerPath}`));
+    } catch (error) {
+        console.error(`Error parsing JSON for headers file ${headerPath}`);
+        return {};
+    }
+}
+
+function getHeaders(path) {
+    return Object.keys(headerMappings).reduce((allHeaders, currentHeaderRegex) => {
```

Comment:
Do not rely on global variables, pass in everything that the function needs as parameters